src/mumble/AudioConfigDialog.cpp

```diff
@@ -497,26 +497,29 @@ void AudioInputDialog::on_qcbTransmit_currentIndexChanged(int v) {
 void AudioInputDialog::on_qcbSystem_currentIndexChanged(int) {
 	qcbDevice->clear();
 
-	QList< audioDevice > ql;
+	QList< audioDevice > choices;
 
 	if (AudioInputRegistrar::qmNew) {
-		AudioInputRegistrar *air = AudioInputRegistrar::qmNew->value(qcbSystem->currentText());
-		ql                       = air->getDeviceChoices();
+		auto air     = AudioInputRegistrar::qmNew->value(qcbSystem->currentText());
+		auto current = air->getDeviceChoice();
+		choices      = air->getDeviceChoices();
 
-		int idx = 0;
+		for (int i = 0; i < choices.size(); ++i) {
+			const auto choice = choices.at(i);
```

Comment:
```suggestion
			const auto &choice = choices.at(i);
```
I guess copying is not needed?

Reply:
It actually gets the object by reference, due to `choices` being a `const` object:

![](https://user-images.githubusercontent.com/5897523/117582579-ec73e000-b102-11eb-9fbf-6191a7779611.png)

However, I'm going to add the ampersand, to make it clearer.

Reply:
It actually gets the object by reference:

![](https://user-images.githubusercontent.com/5897523/117582579-ec73e000-b102-11eb-9fbf-6191a7779611.png)

For reference: https://doc.qt.io/qt-5/qlist.html#at

However, I'm going to add the ampersand, to make it clearer.

Reply:
I incorrectly assumed `auto` would do it automatically, but that's not the case.

![](https://user-images.githubusercontent.com/5897523/117582579-ec73e000-b102-11eb-9fbf-6191a7779611.png)

![](https://user-images.githubusercontent.com/5897523/117582701-97849980-b103-11eb-979c-67bc135cc25c.png)

For reference: https://doc.qt.io/qt-5/qlist.html#at

Reply:
Yeah `const` and `&` do usually have to be typed explicitly

Reply:
I left out `const` because it's implied when the reference is to a `const` object.